test/services/database/QueryStrategies.test.ts

```diff
@@ -0,0 +1,170 @@
+import { Kysely } from "kysely";
+import { IMemoryDb, newDb } from "pg-mem";
+import { beforeEach, describe, expect, it } from "vitest";
+import { QueryStrategy } from "../../../src/services/database/strategies/QueryStrategy.js";
+import type { DataDatabase } from "../../../src/types/kyselySupabaseData.js";
+import { BaseQueryArgsType } from "../../../src/lib/graphql/BaseQueryArgs.js";
+
+type GeneratedAlways<T> = import("kysely").GeneratedAlways<T>;
```

Comment:
What does this type represent?

Reply:
Typedoc: `A shortcut for defining columns that are only database-generated (like postgres GENERATED ALWAYS AS IDENTITY). No insert/update is allowed.`